src/guide/advanced/triggers.md

```diff
@@ -102,12 +102,24 @@ The reason why we add this offset, has to do with Iroha being _best effort_. Ima
 So, the offset is meant to anticipate when the block would get added to the chain, so that people who were just 4 minutes early don't have to be potentially several hours late. Because more triggers get executed sooner, your throughput is also infinitesimally smaller. We could also say "you should aim to execute your trigger slightly earlier than consensus starts", but people writing smartcontracts already have too much to worry about.
 
 
-### Block-based triggers
+### Pre-commit Triggers
+
+This is a variant of timed triggers that gets run before transactions get committed. In reality it triggers in the next block, by leaving a special event, but it is effectively a delayed pre-commit that can track the behaviour of transactions in the pipeline.
+
+::: info
+
+These triggers are not meant for restricting the execution of transactions. If you want to stop your users from transferring more than X amount of Y to user Z, you really want a _permission_. While you could hack the pre-commit triggers to emulate the desired behaviour, this is neither economical in terms of gas fees, nor computation.
+
+Until Iroha 2 supports WASM-based _permissions validators_, however, your only choice are pre-commit triggers.
+
+:::
+
+### Block-based Triggers
 
 Same as timed triggers, but instead of only being interested in the time-stamp, this trigger is interested in the block height. While the mechanism for the triggers is similar the use cases are different.
 
 
-### By-call triggers
+### By-call Triggers
 
 These triggers get executed only once the `CallTrigger(trigger_name)` instruction is executed. These can be useful if you want to achieve dynamic linkage between different smartcontract modules. Space is precious, so want to use as little of it as you can. So you follow the UNIX design philosophy, and instead of creating one large smartcontract, you create many smaller ones, and re-use as much logic as you can.
```

Comment:
```suggestion
These triggers get executed only once the `CallTrigger(trigger_name)` instruction is executed. These can be useful if you want to achieve dynamic linkage between different smartcontract modules. Space is precious, so you want to use as little of it as you can. Therefore, you follow the UNIX design philosophy, and instead of creating one large smartcontract, you create many smaller ones, and re-use as much logic as you can.
```